primary/src/header_waiter.rs

```diff
@@ -263,20 +263,15 @@ impl HeaderWaiter {
                     }
                 },
                 // we poll the availability of a slot to send the result to the core simultaneously
-                (Some(result), permit) = try_fut_and_permit!(waiting.try_next(), self.tx_core) => match result {
-                    Some(header) => {
-                        let _ = self.pending.remove(&header.id);
-                        for x in header.payload.keys() {
-                            let _ = self.batch_requests.remove(x);
-                        }
-                        for x in &header.parents {
-                            let _ = self.parent_requests.remove(x);
-                        }
-                        permit.send(header);
-                    },
-                    None => {
-                        // This request has been canceled.
```

Comment:
I'd like not to lose that comment, perhaps reinsert it on the line of the closing bracket of the destructuring `if let`?